baml_language/crates/baml_hir/src/type_ref.rs

```diff
@@ -84,6 +84,10 @@ impl TypeRef {
     }
 
     /// Create a `TypeRef` from an AST `TypeExpr` node.
+    /// TODO: it seems that type parsing is occurring here!!
+    /// I believe that the parser should be what resolves types syntactically
+    /// this code is brittle (e.g, try the type int?? or int[][]). it will be resolved
+    /// as a type path when it should be either incorrect syntax, or something else.
```

Comment:
Why would `int[][]` be incorrect syntax here? Isn't it just list of lists of `int`.

Then `int??` may seem like incorrect syntax but thinking about type aliases cases like:

```ts
type Example = int?;

type Valid = Example?;
```

I'd say it shouldn't fail.